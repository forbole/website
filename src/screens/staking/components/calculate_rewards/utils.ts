/* eslint-disable no-unused-vars */
import * as R from 'ramda';

type NetworkDataProps = {
  bonded: (data: any) => number;
  inflation: (data: any) => number;
  supply: (data: any) => number;
  commissionRate: (data: any) => number;
  marketPrice: (data: any) => number;
  gecko?: any;
};

type NetworkProps = {
  [key: string]: NetworkDataProps;
};

export const toFixed = (num: number): number => {
  return Number(num?.toFixed(2) ?? '0');
};

export const defaultConverter = (ratio: number) => (num: number) => {
  return num / ratio;
};
export const uAtomToAtom = defaultConverter(1000000);

export const uLunaToLuna = defaultConverter(1000000);

export const uKavaToKava = defaultConverter(1000000);

export const uAktToAkash = defaultConverter(1000000);

export const uBandToBand = defaultConverter(1000000);

export const uIovToIov = defaultConverter(1000000);

export const uEMoneyToEMoney = defaultConverter(1000000);

export const uIrisToIris = defaultConverter(1000000);

export const nanoLikeToLike = defaultConverter(1000000000);

export const uCryptoOrgToCryptoOrg = defaultConverter(100000000);

export const uSentinelToSentinel = defaultConverter(1000000);

// Need adjusting the converter
export const uFetchAIToFetchAI = defaultConverter(1000000000000000000);

// Regen Network not listed on Coingecko yet
export const uRegenToRegen = defaultConverter(1000000);

export const uBitsongToBitsong = defaultConverter(1000000);

export const uOasisToOasis = defaultConverter(1);

// need to adjust the converter
export const uKusamaToKusama = defaultConverter(1);

export const uFlowToFlow = defaultConverter(1);

export const uSolanaToSolana = defaultConverter(1);

export const defaultFunctions = (converter: any) => ({
  bonded: (data: any) => {
    return converter(Number(R.pathOr(0, ['result', 'bonded_tokens'], data)));
  },
  inflation: (data: any) => {
    return toFixed(Number(R.pathOr(0, ['result'], data))) ?? 0;
  },
  supply: (data: any) => {
    return converter(Number(R.pathOr(0, ['result'], data)));
  },
  commissionRate: (data: any) => {
    return Number(
      R.pathOr(0, ['result', 'commission', 'commission_rates', 'rate'], data)
    );
  },
  marketPrice: (data: any) => {
    return toFixed(
      Number(R.pathOr(0, ['market_data', 'current_price', 'usd'], data))
    );
  },
  converter,
});

const cosmos: any = R.clone(defaultFunctions(uAtomToAtom));
cosmos.gecko = 'https://api.coingecko.com/api/v3/coins/cosmos';

const terra: any = R.clone(defaultFunctions(uLunaToLuna));
terra.gecko = 'https://api.coingecko.com/api/v3/coins/terra-luna';

const kava: any = R.clone(defaultFunctions(uKavaToKava));
kava.gecko = 'https://api.coingecko.com/api/v3/coins/kava';

const akash: any = R.clone(defaultFunctions(uAktToAkash));
akash.gecko = 'https://api.coingecko.com/api/v3/coins/akash-network';

const band: any = R.clone(defaultFunctions(uBandToBand));
band.gecko = 'https://api.coingecko.com/api/v3/coins/band-protocol';

const iov: any = R.clone(defaultFunctions(uIovToIov));
iov.gecko = 'https://api.coingecko.com/api/v3/coins/starname';

const likecoin: any = R.clone(defaultFunctions(nanoLikeToLike));
likecoin.gecko = 'https://api.coingecko.com/api/v3/coins/likecoin';

const vsys: any = R.clone(defaultFunctions(uBandToBand));
vsys.gecko = 'https://api.coingecko.com/api/v3/coins/v-systems';

const emoney: any = R.clone(defaultFunctions(uBandToBand));
emoney.gecko = 'https://api.coingecko.com/api/v3/coins/iris-network';

// const iris: any = {
//   bonded: (data: any) => {
//     return Number(R.pathOr(0, ["bonded_tokens"], data));
//   },
//   inflation: (data: any) => {
//     const [inflationData] = data?.filter(
//       (x) => x?.type === "irishub/mint/Params"
//     );
//     return (
//       toFixed(Number(R.pathOr(0, ["value", "inflation"], inflationData))) ?? 0
//     );
//   },
//   supply: (data: any) => {
//     const [supply] = R.pathOr([], ["total_supply"], data).filter(
//       (x) => x.denom === "iris-atto"
//     );
//     return uIrisToIris(Number(R.pathOr(0, ["amount"], supply)));
//   },
//   commissionRate: (data: any) => {
//     return Number(R.pathOr(0, ["commission", "rate"], data));
//   },
//   marketPrice: (data: any) => {
//     return toFixed(
//       Number(R.pathOr(0, ["market_data", "current_price", "usd"], data))
//     );
//   },
// };

const iris: any = R.clone(defaultFunctions(uIrisToIris));
iris.gecko = 'https://api.coingecko.com/api/v3/coins/iris-network';

const cryptoOrg: any = R.clone(defaultFunctions(uCryptoOrgToCryptoOrg));
cryptoOrg.gecko = 'https://api.coingecko.com/api/v3/coins/crypto-com-chain';

const sentinel: any = R.clone(defaultFunctions(uSentinelToSentinel));
sentinel.gecko = 'https://api.coingecko.com/api/v3/coins/sentinel';

const fetchAI: any = R.clone(defaultFunctions(uFetchAIToFetchAI));
fetchAI.gecko = 'https://api.coingecko.com/api/v3/coins/fetch-ai';

// Regen Network not listed on Coingecko yet
const regen: any = R.clone(defaultFunctions(uRegenToRegen));
// regen.gecko = "https://api.coingecko.com/api/v3/coins/regen-network";

const bitsong: any = R.clone(defaultFunctions(uBitsongToBitsong));
bitsong.gecko = 'https://api.coingecko.com/api/v3/coins/bitsong';

const oasis: any = R.clone(defaultFunctions(uOasisToOasis));
oasis.gecko = 'https://api.coingecko.com/api/v3/coins/oasis-network';

const kusama: any = R.clone(defaultFunctions(uKusamaToKusama));
kusama.gecko = 'https://api.coingecko.com/api/v3/coins/kusama';

const flow: any = R.clone(defaultFunctions(uFlowToFlow));
flow.gecko = 'https://api.coingecko.com/api/v3/coins/flow';

const solana: any = R.clone(defaultFunctions(uSolanaToSolana));
solana.gecko = 'https://api.coingecko.com/api/v3/coins/solana';

// available networks for calculations
export const networkFunctions: NetworkProps = {
  cosmos,
  kava,
  akash,
  terra,
  iov,
  likecoin,
  iris,
  band,
  emoney,
  vsys,
  solana,
  'band-protocol': band,
  cryptoOrg,
  sentinel,
  'fetch.ai': fetchAI,
  'regen-network': regen,
  bitsong,
  'oasis-protocol': oasis,
  kusama,
  flow,
};
