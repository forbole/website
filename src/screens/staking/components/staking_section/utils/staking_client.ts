import type { ChainId } from "@src/screens/staking/lib/context";

const baseUrl = process.env.NEXT_PUBLIC_STAKING_API as string;

if (!baseUrl) {
  throw new Error("NEXT_PUBLIC_STAKING_API is not set");
}

const parsedBaseUrl = baseUrl.replace(/\/$/, "");

type Options = {
  body?: FormData | string;
  headers?: Record<string, string>;
  method?: "GET" | "POST";
};

const fetchJson = <A = any>(uri: string, opts?: Options): Promise<A> =>
  fetch(parsedBaseUrl + uri, {
    ...opts,
    headers: {
      ...opts?.headers,
      "Content-Type": "application/json",
    },
  }).then((res) => res.json());

type StakeResponse = {
  tx: {
    authInfo: {
      fee: {
        amount: Array<{ amount: string; denom: string }>;
        gas_limit: string;
      };
    };
    body: {
      memo: string;
      messages: Array<{
        "@type": string;
        "amount": { amount: string; denom: string };
        "delegator_address": string;
        "validator_address": string;
      }>;
    };
  };
};

type GetAddressInfoResponse = {
  balances: { amount: string; denom: string };
};

type GetStakingInfoResponse = {
  rpc: string;
};

export const stakingClient = {
  broadcast: async (network: ChainId, body: unknown) =>
    fetchJson(`/api/broadcast/${network}`, {
      body: JSON.stringify(body),
      method: "POST",
    }),
  claimRewards: async (network: ChainId, address: string) =>
    fetchJson(`/api/claim_rewards`, {
      body: JSON.stringify({
        address,
        network,
      }),
      method: "POST",
    }),
  getAddressInfo: async (network: ChainId, address: string) =>
    fetchJson<GetAddressInfoResponse>(`/api/address/${network}/${address}`),

  getRewardsInfo: async (network: ChainId, address: string) =>
    fetchJson(`/api/rewards/${network}/${address}`),
  getStakingInfo: async (network: ChainId) =>
    fetchJson<GetStakingInfoResponse>(`/api/staking_info/${network}`),
  stake: async (network: ChainId, address: string, amount: string) =>
    fetchJson<StakeResponse>(`/api/stake`, {
      body: JSON.stringify({
        address,
        amount,
        network,
      }),
      method: "POST",
    }),
  unstake: async (network: ChainId, address: string, amount: string) =>
    fetchJson(`/api/unstake`, {
      body: JSON.stringify({
        address,
        amount,
        network,
      }),
      method: "POST",
    }),
};
