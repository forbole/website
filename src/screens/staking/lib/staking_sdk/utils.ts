import type { WalletId } from "./types";

const localStorageKey = "connectedWallets";

export const getConnectedWallets = (): WalletId[] => {
  const connectedWallets = window.localStorage.getItem(localStorageKey);

  if (!connectedWallets) {
    return [];
  }

  try {
    const parsedWallets = JSON.parse(connectedWallets);

    return parsedWallets;
  } catch (e) {
    return [];
  }
};

export const addToConnectedWallets = (wallet: WalletId) => {
  const connectedWallets = getConnectedWallets();

  const newWalletsSet = new Set([...connectedWallets, wallet]);
  const newWallets = Array.from(newWalletsSet);

  window.localStorage.setItem(localStorageKey, JSON.stringify(newWallets));
};

export const clearConnectedWallets = () => {
  window.localStorage.removeItem(localStorageKey);
};
