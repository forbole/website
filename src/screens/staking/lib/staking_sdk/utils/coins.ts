import BigNumber from "bignumber.js";

import type { Coin } from "../core/base";
import { CoinDenom, StakingNetworkId } from "../core/base";

export const networkToUnnormalisedDenom = {
  [StakingNetworkId.Akash]: "UAKT",
  [StakingNetworkId.Celestia]: "UTIA",
  [StakingNetworkId.CelestiaTestnet]: "UTIA",
  [StakingNetworkId.ComposableFinance]: "PPICA",
  [StakingNetworkId.CosmosHub]: "UATOM",
  [StakingNetworkId.CosmosHubTestnet]: "UATOM",
  [StakingNetworkId.DyDx]: "ADYDX",
  [StakingNetworkId.Dymension]: "ADYM",
  [StakingNetworkId.Evmos]: "AEVMOS",
  [StakingNetworkId.Injective]: "inj", // This is important to be lowercase, since inj != INJ
  [StakingNetworkId.IslamicCoin]: "AISLM",
  [StakingNetworkId.Kava]: "UKAVA",
  [StakingNetworkId.KavaTestnet]: "UKAVA",
  [StakingNetworkId.Osmosis]: "UOSMO",
  [StakingNetworkId.Stargaze]: "USTARS",
  [StakingNetworkId.StargazeTestnet]: "USTARS",
} as const satisfies Record<StakingNetworkId, string>;

type DenomToNormalise = (typeof networkToUnnormalisedDenom)[StakingNetworkId];

const uExp = 6;
const pExp = 12;
const aExp = 18;
const exp0 = 0;

const denomMap: Record<DenomToNormalise, [CoinDenom, number]> = {
  ADYDX: [CoinDenom.DYDX, aExp],
  ADYM: [CoinDenom.DYM, aExp],
  AEVMOS: [CoinDenom.EVMOS, aExp],
  AISLM: [CoinDenom.ISLM, aExp],
  // Because inj != INJ, this needs to keep the lower case, it is handled when
  // normalising the coin
  inj: [CoinDenom.INJ, exp0],
  PPICA: [CoinDenom.PICA, pExp],
  UAKT: [CoinDenom.AKT, uExp],
  UATOM: [CoinDenom.ATOM, uExp],
  UKAVA: [CoinDenom.KAVA, uExp],
  UOSMO: [CoinDenom.OSMO, uExp],
  USTARS: [CoinDenom.STARS, uExp],
  UTIA: [CoinDenom.TIA, uExp],
};

export const normaliseCoin = (coin: Coin): Coin => {
  // Special case where INJ != inj
  if (coin.denom === "inj") {
    return {
      amount: new BigNumber(coin.amount)
        .div(new BigNumber(10).pow(18))
        .toString(),
      denom: CoinDenom.INJ,
    };
  }

  const compared = coin.denom?.toUpperCase();

  const parseNum = (exp: number) =>
    new BigNumber(coin.amount).dividedBy(new BigNumber(10).pow(exp)).toString();

  const [denom, exp] = denomMap[compared as DenomToNormalise] || [
    coin.denom,
    0,
  ];

  if (!denom) {
    throw new Error(`Unknown denom ${coin}`);
  }

  let { amount } = coin;

  if (compared !== denom.toUpperCase()) {
    amount = parseNum(exp);
  }

  return {
    amount,
    denom: denom.toUpperCase(),
  };
};

export const normaliseDenom = (denom: string): string =>
  normaliseCoin({ amount: "0", denom }).denom;

export const getEmptyCoin = (denom = ""): Coin => ({ amount: "0", denom });

export const sumCoins = (coinA?: Coin, coinB?: Coin): Coin =>
  [coinA, coinB]
    .filter((c): c is Coin => !!c?.denom)
    .map((c) => normaliseCoin(c))
    .reduce((acc, coin) => {
      const { amount } = acc;
      const { amount: amountB } = coin;

      if (acc.denom && coin.denom && acc.denom !== coin.denom) {
        console.error("Unexpected sum of different denoms", acc, coin);

        // Don't throw here for now
        return acc;
      }

      const newAmount = new BigNumber(amount).plus(amountB).toString();
      const denom = [acc.denom, coin.denom].find(Boolean) || "";

      return {
        amount: newAmount,
        denom,
      };
    }, getEmptyCoin());

export const sumAllCoins = (coins: Coin[]): Coin =>
  (coins || []).reduce((acc, coin) => sumCoins(acc, coin), getEmptyCoin());
