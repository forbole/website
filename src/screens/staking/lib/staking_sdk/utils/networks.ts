import type { Network, NetworkKey } from "@src/utils/network_info";

import type { NetworkInfo } from "../core";
import { networkKeyToNetworkId } from "../core";

export const sortNetworksByName = (a: Network, b: Network) => {
  const networkIdA = networkKeyToNetworkId[a.key as NetworkKey];
  const networkIdB = networkKeyToNetworkId[b.key as NetworkKey];

  if (networkIdA && !networkIdB) {
    return -1;
  }

  if (!networkIdA && networkIdB) {
    return 1;
  }

  return a.name.localeCompare(b.name);
};

export const getUnbondingTimeForNetwork = (
  networkInfo: NetworkInfo | null,
  locale?: string,
) => {
  if (!networkInfo) {
    return null;
  }

  const { unbonding_period: unbondingPeriod } = networkInfo;

  if (!unbondingPeriod) {
    return null;
  }

  const now = new Date();
  const unbondingPeriodSeconds = parseInt(unbondingPeriod, 10);
  const days = Math.ceil(unbondingPeriodSeconds / 86400);
  const nextDate = new Date(now.getTime() + unbondingPeriodSeconds * 1000);

  const nextDateStr = nextDate.toLocaleDateString(locale, {
    day: "numeric",
    month: "long",
    year: "numeric",
  });

  return {
    date: nextDateStr,
    days,
  };
};
