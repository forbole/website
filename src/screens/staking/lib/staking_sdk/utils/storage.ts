import { walletsSupported } from "../core";
import type { WalletId } from "../core/base";

const walletsStorageKey = "connectedWallets";
const stakedStorageKey = "hasStaked";
const unstakedStorageKey = "hasUnstaked";

const solanaWithdrawReady = "solanaWithdrawReady";
const solflareCloseNotifiedKey = "solflareCloseNotified";

const isValInLastDay = (val?: null | string) => {
  if (!val) return false;

  const dateTimestamp = JSON.parse(val);

  if (Number.isNaN(dateTimestamp)) return false;

  const now = Date.now();

  return now - dateTimestamp < 1000 * 60 * 60 * 24;
};

export const getConnectedWallets = (): WalletId[] => {
  const connectedWallets = window.localStorage.getItem(walletsStorageKey);

  if (!connectedWallets) {
    return [];
  }

  try {
    const parsedWallets: WalletId[] = JSON.parse(connectedWallets);

    return parsedWallets.filter((w) => walletsSupported.has(w));
  } catch (e) {
    return [];
  }
};

export const addToConnectedWallets = (wallet: WalletId) => {
  const connectedWallets = getConnectedWallets();

  const newWalletsSet = new Set([...connectedWallets, wallet]);
  const newWallets = Array.from(newWalletsSet);

  window.localStorage.setItem(walletsStorageKey, JSON.stringify(newWallets));
};

export const setConnectedWallet = (walletsIds: WalletId[]) => {
  if (!walletsIds.length) {
    window.localStorage.removeItem(walletsStorageKey);
  } else {
    const newWalletsSet = new Set([...walletsIds]);
    const newWallets = Array.from(newWalletsSet);

    window.localStorage.setItem(walletsStorageKey, JSON.stringify(newWallets));
  }
};

export const getHasStaked = () =>
  window.localStorage.getItem(stakedStorageKey) === "true";

export const setHasStaked = () =>
  window.localStorage.setItem(stakedStorageKey, "true");

export const getHasUnstaked = () =>
  window.localStorage.getItem(unstakedStorageKey) === "true";

export const setHasUnstaked = () =>
  window.localStorage.setItem(unstakedStorageKey, "true");

/**
 * This function checks if the value was notified in the last 24 hours
 */
export const getSolanaWithdrawNotified = () => {
  const val = window.localStorage.getItem(solanaWithdrawReady);

  return isValInLastDay(val);
};

export const setSolanaWithdrawNotified = () => {
  window.localStorage.setItem(solanaWithdrawReady, JSON.stringify(Date.now()));
};

export const getSolflareCloseNotified = () => {
  const val = window.localStorage.getItem(solflareCloseNotifiedKey);

  return isValInLastDay(val);
};

export const setSolflareCloseNotified = () => {
  window.localStorage.setItem(
    solflareCloseNotifiedKey,
    JSON.stringify(Date.now()),
  );
};
