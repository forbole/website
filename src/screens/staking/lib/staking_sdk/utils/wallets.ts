import type { WalletId } from "../core";

export const sortWallets = (a: WalletId, b: WalletId) => a.localeCompare(b);
