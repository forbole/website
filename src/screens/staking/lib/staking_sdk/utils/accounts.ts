import type { Coin } from "@cosmjs/stargate";
import BigNumber from "bignumber.js";

import { getNetworkInfo } from "@src/utils/network_info";

import type { Account } from "../core";
import { networkIdToNetworkKey, testnetNetworks } from "../core";
import { normaliseCoin, sumAllCoins, sumCoins } from "./coins";
import { sortNetworksByName } from "./networks";

export const accountHasDelegations = (account?: Account): boolean =>
  !!account?.info?.delegation &&
  Object.keys(account.info.delegation).length > 0;

export const accountHasRewards = (account?: Account): boolean =>
  !!account?.rewards &&
  Array.isArray(account.rewards) &&
  account.rewards.length > 0;

export const filterUniqueAddresses = () => {
  const usedAddresses = new Set<string>();

  return (account: Account) => {
    if (usedAddresses.has(account.address + account.networkId)) {
      return false;
    }

    usedAddresses.add(account.address + account.networkId);

    return true;
  };
};

export const filterOutTestnets = (account: Account) =>
  !testnetNetworks.has(account.networkId);

type NormalisedInfo = { coin: Coin; num: BigNumber } | null;

export const getAccountNormalisedDelegation = (
  account?: Account,
): NormalisedInfo => {
  const delegationProp = account?.info?.delegation;

  const delegation = Array.isArray(delegationProp)
    ? sumAllCoins(delegationProp)
    : delegationProp;

  if (!delegation?.denom) return null;

  const available = normaliseCoin(delegation);

  return {
    coin: available,
    num: new BigNumber(available.amount),
  };
};

export const getAccountNormalisedBalance = (
  account?: Account,
): NormalisedInfo => {
  if (!account?.info?.balances) return null;

  const balance = normaliseCoin(account.info.balances);

  return {
    coin: balance,
    num: new BigNumber(balance.amount),
  };
};

export const sortAccounts = (a: Account, b: Account) => {
  if (a.networkId !== b.networkId) {
    return a.networkId.localeCompare(b.networkId);
  }

  return a.address.localeCompare(b.address);
};

export const getClaimableRewardsForAccount = (start: Coin, account: Account) =>
  (Array.isArray(account.rewards) ? account.rewards : []).reduce(
    (acc2, reward) => {
      if (start.denom?.toUpperCase() === reward.denom?.toUpperCase()) {
        return sumCoins(acc2, reward);
      }

      return acc2;
    },
    start,
  );

export const sortAccountsByNetworkName = (a: Account, b: Account) => {
  if (a.networkId === b.networkId) {
    return a.address.localeCompare(b.address);
  }

  const networkAKey = networkIdToNetworkKey[a.networkId];
  const networkBKey = networkIdToNetworkKey[b.networkId];

  const networkA = getNetworkInfo(networkAKey);
  const networkB = getNetworkInfo(networkBKey);

  return sortNetworksByName(networkA, networkB);
};
