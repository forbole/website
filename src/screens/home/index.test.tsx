import React from 'react';
import renderer from 'react-test-renderer';
import { wait } from '@tests/utils';
import Home from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Layout: (props: any) => <div id="Layout" {...props} />,
}));

jest.mock('./components', () => ({
  HomeAnimation: (props: any) => <div id="HomeAnimation" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Home', () => {
  it('matches snapshot', async () => {
    let component: any;
    renderer.act(() => {
      component = renderer.create(<Home />);
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
