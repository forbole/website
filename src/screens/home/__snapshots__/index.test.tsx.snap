// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home matches snapshot 1`] = `
<div
  description="description"
  id="Layout"
  image="/images/assets/Facebook-Forbole.png"
  navLink="/"
  title="forbole"
  twitterImage="/images/assets/Twitter-Forbole.png"
>
  <div
    className="MuiBox-root css-17ojfhh"
  >
    <div
      className="MuiBox-root css-rbdyf8"
    />
    <div
      className="MuiBox-root css-1vud5kq"
    >
      <div
        className="MuiBox-root css-1p1g7i8"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 css-7so9kd-MuiTypography-root"
      >
        titleD
      </p>
      <div
        className="MuiBox-root css-xt8px5"
      />
      <div
        className="MuiBox-root css-xyedca"
      />
    </div>
  </div>
</div>
`;
